### DISCUSSION TITLE
Help with automatic rotation?

### DISCUSSION THREAD

**Discussion by jmaris** [OpenPrinting/cups-filters#612] [Q&A]
Hi, 

I'm not sure if this is the right place for this, but if it is I'd really appreciate some help. I have a Brother MFC-J1010DW set up with IPP Everywhere.

In the past, when I wanted to print slides from a presentation, I didn't have to change any settings, the document would automatically be printed in Landscape.

Now when I try to print, it prints a cropped version of the slide onto the paper in portrait.

Is there a way I can get CUPS to automatically print landscape documents as landscape?

Thanks

**Comment by tillkamppeter** [comment]
Please make available to us the file you want to print.

Also tell us how you have printed it and with which settings.

Also please attach the PPD file of the print queue (`/etc/cups/ppd/*.ppd`)

Please also put CUPS into debug mode
```
cupsctl --debug-logging
```
send the faulty print job again and after that, attch the file
```
/var/log/cups/error_log
```
Note that for attaching the PPD and log files here you need to rename them to have the file name extension `.txt`.

**Reply by jmaris** [reply]
Hi, thanks for the rapid reply!

There were two PPD files in the directory, I've attached both with their original names.

Even after putting CUPS into debug mode with cupsctl, there is no errror_log in the /var/log/cups folder (the folder exists but is empty). I have tried to set it to debug mode with and without sudo. After running it, cupsctl returns `_debug_logging=1` so I assume it was correctly set.

I've also attached the document, and a photo of how it prints.

Please let me know if there is anything else you need!

![output](https://github.com/user-attachments/assets/781f0538-05fb-4bcf-82c2-e20902ed4732)
[printed file.pdf](https://github.com/user-attachments/files/18274333/printed.file.pdf)
[MFC-J1010DW.ppd.O.txt](https://github.com/user-attachments/files/18274334/MFC-J1010DW.ppd.O.txt)
[MFC-J1010DW.ppd.txt](https://github.com/user-attachments/files/18274335/MFC-J1010DW.ppd.txt)

**Comment by tillkamppeter** [comment]
I am not able to reproduce the bug by commands to run the filters manually:
```
cat printed.file.pdf | PPD=MFC-J1010DW.ppd.txt CONTENT_TYPE=application/pdf FINAL_CONTENT_TYPE=image/urf /usr/lib/cups/filter/pdftopdf 1 1 1 1 '' > out.pdf
cupsfilter -P MFC-J1010DW.ppd.txt -m application/vnd.cups-pdf printed.file.pdf > out.pdf
cupsfilter -P MFC-J1010DW.ppd.txt -m image/urf printed.file.pdf > out.urf
```

Which distro are you using? Which version of cups-filters is used?

For your `error_log` file not getting written, could you check whether there are any messages related to it in the `/var/log/syslog` (or in the journal)? For example any `DENIED` messages from AppArmor.

Also, could you attach your files `/etc/cups/cupsd.conf` and `/etc/cups/cups-files.conf`?

Note that the `.O` files written by CUPS are only backups of earlier versions of the original files and have no relevance for the functionality of the CUPS queues.

**Reply by jmaris** [reply]
Hi,

Regrettably, I ran the commands you used to reproduce the bug, and it automatically rotated the document to the correct orientation. I assume this means that the issue is not with CUPS-filters, but I've attached the requested things to the extent possible, in case i'm missing something.

Just for information: I'm running fedora with CUPS version 2.4.11. I assume this means that the logs are instead written to the systemd journal. After having set cups debug on, I searched the systemd journal. The only thing I saw was the following spammed in the logs, but this happened all the time (not when printing) and there were no specific messages when printing.
```
Jan 04 13:04:37 claudiapc cupsd[11199]: cupsdSetBusyState: newbusy="Active clients", busy="Not busy"
Jan 04 13:04:37 claudiapc cupsd[11199]: [Client 47654] Server address is "/run/cups/cups.sock".
Jan 04 13:04:37 claudiapc cupsd[11199]: [Client 47654] skipping getpeercon()
Jan 04 13:04:37 claudiapc cupsd[11199]: [Client 47654] Accepted from localhost (Domain)
Jan 04 13:04:37 claudiapc cupsd[11199]: [Client 47654] Waiting for request.
Jan 04 13:04:37 claudiapc cupsd[11199]: cupsdSetBusyState: newbusy="Active clients", busy="Active clients"
Jan 04 13:04:37 claudiapc cupsd[11199]: [Client 47655] Server address is "/run/cups/cups.sock".
Jan 04 13:04:37 claudiapc cupsd[11199]: [Client 47655] skipping getpeercon()
Jan 04 13:04:37 claudiapc cupsd[11199]: [Client 47655] Accepted from localhost (Domain)
Jan 04 13:04:37 claudiapc cupsd[11199]: [Client 47655] Waiting for request.
Jan 04 13:04:37 claudiapc cupsd[11199]: [Client 47654] HTTP_STATE_WAITING Closing for error 32 (Broken pipe)
Jan 04 13:04:37 claudiapc cupsd[11199]: [Client 47654] Closing connection.
```

I also atttached the configuration files as requested:

[cupsd.conf.txt](https://github.com/user-attachments/files/18306766/cupsd.conf.txt)

[cups-files.conf.txt](https://github.com/user-attachments/files/18306765/cups-files.conf.txt)

If you have any clues i'd really appreciate it!

**Comment by pdfkungfoo** [comment]
Are you, by any chance, printing the file from a GUI application?
Is this application, by any chance, written with the Qt toolkit?

Then the pull request being discussed here: 

*  https://codereview.qt-project.org/c/qt/qtbase/+/624835

probably is related to your problem.
